tests/storage/test_storage.rs

```diff
@@ -578,6 +599,31 @@ fn test_txn_store_rawkv() {
     store.raw_get_ok(b"key".to_vec(), None);
 }
 
+#[test]
+fn test_txn_store_lock_primary() {
+    let store = new_assertion_storage();
+    // txn1 locks "p" then aborts.
+    store.prewrite_ok(vec![Mutation::Put((make_key(b"p"), b"p1".to_vec()))],
+                      b"p",
+                      1);
+
+    // txn2 wants to write "p", "s".
+    store.prewrite_locked(vec![Mutation::Put((make_key(b"p"), b"p2".to_vec())),
```

Comment:
Do you mean start a new txn to write only "s"?
It's ok for other txns to write "s" when it's not locked.